src/Features/Core/Portable/SimplifyInterpolation/AbstractSimplifyInterpolationDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.VirtualChars;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.SimplifyInterpolation
+{
+    internal abstract class AbstractSimplifyInterpolationDiagnosticAnalyzer<
+        TInterpolationSyntax,
+        TExpressionSyntax> : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+        where TInterpolationSyntax : SyntaxNode
+        where TExpressionSyntax : SyntaxNode
+    {
+        protected AbstractSimplifyInterpolationDiagnosticAnalyzer()
+           : base(IDEDiagnosticIds.SimplifyInterpolationId,
+                  CodeStyleOptions.PreferSimplifiedInterpolation,
+                  new LocalizableResourceString(nameof(FeaturesResources.Simplify_interpolation), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
```

Comment:
Can you put the localizable resource strings as static readonly fields? Would make it easier to read the constructor/base

Reply:
i can.  though this pattern is very universal for our analyzers.

Reply:
going to keep it like this.  we use this *everywhere* :)